### PR TITLE
add imported package names shadowing linter

### PR DESCRIPTION
**Pull request by NexoMichael**
Sometimes developers use the same name for the variables as imported packages, more often I see something like importing "log" package and having "log" variable in the code.

This linter checks what imports the package has and what variables got created and compares one to another.

### CODE REVIEW

**Comment by Quasilyte** [comment]
My favorite case is `path/filepath` and `path`.
They make me nervous when naming something that holds file path. :sweat_smile:

File: lint/importShadow_checker.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"go/ast"
+	"go/types"
+
+	"github.com/go-critic/go-critic/lint/internal/astwalk"
+)
+
+func init() {
+	addChecker(&importShadowChecker{}, attrSyntaxOnly)
```
Comment (Quasilyte) [inline lint/importShadow_checker.go]: Since this checker uses `types.Package`, it should probably not have `attrSyntaxOnly` attribute.
Generally, it's safer to omit this attribute, at least until we implement #231.

File: lint/importShadow_checker.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+type importShadowChecker struct {
+	checkerBase
+}
+
+func (c *importShadowChecker) Init() {}
```
Comment (Quasilyte) [inline lint/importShadow_checker.go]: You may remove `Init` method as this checker has no state that needs to be initialized.

**Comment by fexolm** [comment]
Please add `attrExperimental` to you checker. We always mark new checkers as experemental.

File: lint/importShadow_checker.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
+package lint
+
+//! Detects when imported package names shadowed in assignments.
+//
+// @Before:
+// import (
+// 	   "fmt"
```
Comment (Quasilyte) [inline lint/importShadow_checker.go]: Looks like there is an extra whitespace here.

Comment (NexoMichael) [reply lint/importShadow_checker.go]: yeap, but extra white space will need some fix in docs generation, will look what is going on there

Comment (NexoMichael) [reply lint/importShadow_checker.go]: let's create additional issue for that, it seems like current approach with
```
for *index < len(lines) && len(lines[*index]) > 0 {
```
for finding description parts will not work for code with empty lines

**Comment by Quasilyte** [comment]
Note: we're following golang/go guidelines for commit messages.
https://golang.org/doc/contribute.html#commit_messages

tl;dr: it should be like: `lint: add importShadow linter`
And the body usually contains either details or `Updates #XXX`/`Fixes #XXX` to provide more context.

File: lint/importShadow_checker.go (hunk 3 above)
Comment (fexolm) [reply lint/importShadow_checker.go]: Its how our docs works. Empty lines in code are not allowed.

**Comment by Quasilyte** [comment]
@NexoMichael, you can use your checker either with `-withExperimental` option that turns on all experimental checkers for `-enable=all` or with explicit list like `-enable=importShadow`.

File: lint/importShadow_checker.go (hunk 3 above)
Comment (Quasilyte) [reply lint/importShadow_checker.go]: This is mostly because we want:
1. Encourage minimal examples.
2. Avoid hard to read doc comments inside checkers source files.

I would use example like this:
```go
// "math" and "fmt" packages are imported.
func f(math int) int {
  fmt := math+1
  return fmt
}
```
Or even something that does not include surrounding func.
